### PR TITLE
Improve chat dialog spacing and layout

### PR DESCRIPTION
**Pull request by nkuehn**
The chat dialog's slightly unpolished look has always bugged my given the eyes that are currently on it.  

Here's a few tweaks that adress my personal main scratch points

- sligthly wider right hand column,  it turns out we have to give it more focus because users start using the AI instead of search 
-  remove duplicated spacing in the resources list 
-  make the resources list scroll inside the rounded box, not the rounded box scroll inside an invisible box (not sure why this slipped through initially)
- align the bottom content elements with the invisible dialog content area (no extra bottom spacing on top of the dialog's own spacing)
- take back 8px of duplicate spacing at the top (not sure why the dialog has its spacing and then another spacing on top of the dialog - hack through negative margin, did not want to touch the dialog proper
-

### CODE REVIEW

**Comment by zbalek** [comment]
Hi @nkuehn - I'm okay to approve the changes but the hint in the text field does not sound correct and it seems that we just accidentally made a mistake there

**Comment by nkuehn** [comment]
@zbalek  Thanks for checking and catching the accidental text change.  I'll revert it. 

this is not trying to replace other work you do - I see it rather as polish we would normally have done with the initial release aready if we had more time back then.   Another factor is that we have increasingly long microsite and page names and the sidebar layout got really hard to read with these. 

Apropos copywriting:  The "Tip": link at the bottom is rarely used but we have actual problems with people changing the subject too much in a given conversation.   Can you imagine a copy that is more explicit?  "click here" is what I would want to say but it's sounding bad.

**Comment by zbalek** [comment]
@nkuehn RE Tip: Perhaps the issue now is that people do not realise that it is in fact a CTA.

I'd suggest two options here:

A) A[ flat button component](https://uikit.commercetools.com/?path=/story/components-buttons--flatbutton) and adding a refresh icon on the left. In terms of copy, below are some suggestions that could work better.

B) Having it as a link, i.e. underlining the text + adding the refresh icon to capture the attention when a user scans through the page.

Personally, I'm leaning towards the se

----------

image

**Review by timonrey — COMMENTED** [review]
This is how the dialog looks like on wider screens. Is that intentional?
Screenshot 2024-07-25 at 18 58 03

Same here: I think the grid template for largeDesktop needs to be adjusted.
Screenshot 2024-07-25 at 18 59 46

**Review by gabriele-ct — APPROVED** [review]

File: packages/gatsby-theme-docs/src/modules/ai-assistant/components/chat-modal-css-components.js [hunk 1]
```diff
@@ -117,6 +118,9 @@ export const SideTopContainer = styled.div`
   flex-grow: 1;
   overflow-y: auto;
   padding: 0px ${designSystem.dimensions.spacings.m};
+  border: 1px solid ${designSystem.colors.light.borderPrimary};
+  background-color: ${designSystem.colors.light.surfacePrimary};
+  border-radius: 4px;
```
[the remark this answers is not in the document]
Comment (nkuehn) [reply packages/gatsby-theme-docs/src/modules/ai-assistant/components/chat-modal-css-components.js:123]: Not convinced to use a token whose name refers specifically to the beta flat for other purposes just because it's the same value.   Would lead to hidden side effects if someone later wants to change the beta flag and in good trust believes the naming.

**Comment by nkuehn** [comment]
> This is how the dialog looks like on wider screens. Is that intentional? 

Yes, columns with text content to read should never get wider than a line length that the human eye can easily track.   But that it gets _smaller_ on large screens is clearly a bug, thank you for spotting that! 

ideally the modal itself would never get as wide at all though but I think @gabriele-ct  once told me that something limits our control over it. True  still?

**Comment by gabriele-ct** [comment]
> > This is how the dialog looks like on wider screens. Is that intentional?
> 
> Yes, columns with text content to read should never get wider than a line length that the human eye can easily track. But that it gets _smaller_ on large screens is clearly a bug, thank you for spotting that!
> 
> ideally the modal itself would never get as wide at all though but I think @gabriele-ct once told me that something limits our control over it. True still?

`FormDialog` component accept a prop `size`. After trying some of the options available I came to the conclusion that the only value which is usable for out purposes is `scale` which adapts the size of the modal to fill the  whole avaliable space. In other cases we get along with `l` for example the chat feedback and the profile modals. 
I had tried that value but, it was too small for the content we wanted to display.

So, in short yes, we still have that limitation, and maybe we could explore other options, like for example use a [drawer component](https://docs.commercetools.com/merchant-center-customizations/api-reference/commercetools-frontend-application-components#component-for-drawer) for the ai assistant.

**Comment by nkuehn** [comment]
@gabriele-ct  thank you!  Let's live with the whitespace at the sides for now then.  Mid term I would rather like to research a sidebar that is not modal so that it becomes possible to navigate the content with the assistant staying open (would only work without separately built microsites)

I pushed a commit that removes the separate handling of the wide screen with spacer elements - spacer elements are not needed, the container has a max width and is auto centered. 

@zbalek  I'l leave the restart text as is to not overscope this change before going on vacation.  I think we need to address both the visual as well as the language copy of the CTA.  To make the language clearly a CTA we may have to separate out the explanation,  e.g. the CTA being just "Start a new Conversation"  and visually different the explanation "Gives better results when changing the topic".

**Review by timonrey — COMMENTED** [review]
The logged-in state looks good now 🙌🏼 
But there's still a bug in the wide layout when not logged-in as you can see here:
Screenshot 2024-07-31 at 08 09 56

**Comment by nkuehn** [comment]
@timonrey  sorry and thank you - my git skills got worse quite fast.  Missed half of the files somehow (should stop using stashing  partial changes if I don't get it right...).

**Review by timonrey — APPROVED** [review]
Thank you! 🙌🏼
